src/storage/src/hummock/store/version.rs

```diff
@@ -610,20 +611,26 @@ impl HummockVersionReader {
                 );
 
                 let mut sstables = vec![];
-                for sstable_info in &level.table_infos[start_table_idx..=end_table_idx] {
-                    if sstable_info
-                        .table_ids
-                        .binary_search(&read_options.table_id.table_id)
-                        .is_err()
-                    {
-                        continue;
-                    }
-                    let sstable = self
-                        .sstable_store
-                        .sstable(sstable_info, &mut local_stats)
+                let fetch_meta_req = level.table_infos[start_table_idx..=end_table_idx]
+                    .iter()
+                    .filter(|sstable_info| {
+                        sstable_info
+                            .table_ids
+                            .binary_search(&read_options.table_id.table_id)
+                            .is_ok()
+                    })
+                    .collect_vec();
+                let fetch_meta_resp = try_join_all(fetch_meta_req.iter().map(|sstable_info| {
+                    self.sstable_store
+                        .sstable_syncable(sstable_info, &local_stats)
                         .in_span(Span::enter_with_local_parent("get_sstable"))
-                        .await?;
-
+                }))
+                .await?;
+                for (sstable_info, (sstable, local_cache_meta_block_miss)) in fetch_meta_req
```

Comment:
(not related to this PR) during the review I noticied that we pass `SstableInfo` instead of `TableHolder` for `ConcatIterator::new` in L648, which will cause a unneccessary meta cache lookup during the ConcatIterator execution. I think we can also optimize it.